src/protocol/bigreq.rs

```diff
@@ -42,88 +32,20 @@ fn major_opcode<Conn: RequestConnection + ?Sized>(conn: &Conn) -> Result<u8, Con
     Ok(info.major_opcode)
 }
 
-/// Opcode for the Enable request
-pub const ENABLE_REQUEST: u8 = 0;
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct EnableRequest;
-impl EnableRequest {
-    /// Serialize this request into bytes for the provided connection
-    fn serialize_impl(self, major_opcode: u8) -> BufWithFds<PiecewiseBuf<'static>> {
-        let length_so_far = 0;
-        let mut request0 = vec![
-            major_opcode,
-            ENABLE_REQUEST,
-            0,
-            0,
-        ];
-        let length_so_far = length_so_far + request0.len();
-        assert_eq!(length_so_far % 4, 0);
-        let length = u16::try_from(length_so_far / 4).unwrap_or(0);
-        request0[2..4].copy_from_slice(&length.to_ne_bytes());
-        (vec![request0.into()], vec![])
-    }
-    pub fn send<Conn>(self, conn: &Conn) -> Result<Cookie<'_, Conn, EnableReply>, ConnectionError>
-    where
-        Conn: RequestConnection + ?Sized,
-    {
-        let (bytes, fds) = self.serialize_impl(major_opcode(conn)?);
-        let slices = bytes.iter().map(|b| IoSlice::new(&*b)).collect::<Vec<_>>();
-        conn.send_request_with_reply(&slices, fds)
-    }
-    /// Parse this request given its header, its body, and any fds that go along with it
-    pub fn try_parse_request(header: RequestHeader, value: &[u8]) -> Result<Self, ParseError> {
-        if header.minor_opcode != ENABLE_REQUEST {
-            return Err(ParseError::InvalidValue);
-        }
-        let _ = value;
-        Ok(EnableRequest
-        )
-    }
-}
-impl Request for EnableRequest {
-    type Reply = EnableReply;
-
-    const EXTENSION_NAME: Option<&'static str> = Some(X11_EXTENSION_NAME);
-
-    fn serialize(self, major_opcode: u8) -> BufWithFds<Vec<u8>> {
-        let (bufs, fds) = self.serialize_impl(major_opcode);
-        // Flatten the buffers into a single vector
-        let buf = bufs.iter().flat_map(|buf| buf.iter().copied()).collect();
-        (buf, fds)
-    }
+fn send_enable<'c, Conn>(req: EnableRequest, conn: &'c Conn) -> Result<Cookie<'c, Conn, EnableReply>, ConnectionError>
```

Comment:
Why split this part into a `send_*` function?

Reply:
It was already a `send` function before, but it was an associated function.
```rust
impl EnableRequest {
    pub fn send<Conn>(self, conn: &Conn) -> Result<Cookie<'_, Conn, EnableReply>, ConnectionError>
    where
        Conn: RequestConnection + ?Sized,
    {
        let (bytes, fds) = self.serialize_impl(major_opcode(conn)?);
        let slices = bytes.iter().map(|b| IoSlice::new(&*b)).collect::<Vec<_>>();
        conn.send_request_with_reply(&slices, fds)
    }
}
```
In "the new world", `EnableRequest` comes from another crate (`x11rb-protocol`) and thus `x11rb` cannot add a function to it. This is why I turned this into `send_enable()` (the rename is to avoid name conflicts with the other `send` functions).

I agree that this is weird and should be changed. Somehow. However, I tried to do the minimum number of changes to get things to compile.

Random quick thought for a next step would be: Get rid of these useless `send_enable` function. With this, we lose the feature to send requests from a `struct` (I had to make a change to this effect in `src/image.rs`). To get this feature back, some new API on connection would be needed, something like `send_with_reply_from_trait<R>(&self, request: R) where R: ReplyRequest`.

Reply:
I'm fine with leaving it as is for now.